File: 05_Systems/Scripts/start-localhost-server.js
/**
 * ---
 * Title: High-Speed Node Localhost Dev Server (SYS-035)
 * ID: SYS-035
 * Type: Script (Node.js HTTP Server)
 * Module: 05_Systems/Scripts
 * BU: ZK Revenue Ops
 * Status: Active
 * Version: 2.0
 * Created: 2026-08-08
 * Updated: 2026-08-08
 * Owner: Zubair (zubairisa10@gmail.com)
 * ---
 */

const http = require('http');
const fs   = require('fs');
const path = require('path');

const PORT = 8080;
const ROOT = path.resolve(__dirname, '../../');

const MIME_TYPES = {
    '.html': 'text/html; charset=UTF-8',
    '.css':  'text/css; charset=UTF-8',
    '.js':   'application/javascript; charset=UTF-8',
    '.json': 'application/json; charset=UTF-8',
    '.png':  'image/png',
    '.jpg':  'image/jpeg',
    '.svg':  'image/svg+xml',
    '.csv':  'text/csv; charset=UTF-8',
};

const server = http.createServer((req, res) => {
    // Disable browser caching completely for dev mode
    res.setHeader('Cache-Control', 'no-store, no-cache, must-revalidate, proxy-revalidate');
    res.setHeader('Pragma', 'no-cache');
    res.setHeader('Expires', '0');

    let reqPath = req.url.split('?')[0];
    if (reqPath === '/') reqPath = '/index.html';

    const filePath = path.join(ROOT, reqPath);

    fs.stat(filePath, (err, stats) => {
        if (err || !stats.isFile()) {
            res.writeHead(404, { 'Content-Type': 'text/plain' });
            res.end(`404 Not Found: ${reqPath}`);
            return;
        }

        const ext  = path.extname(filePath).toLowerCase();
        const mime = MIME_TYPES[ext] || 'application/octet-stream';

        res.writeHead(200, { 'Content-Type': mime });
        fs.createReadStream(filePath).pipe(res);
    });
});

server.listen(PORT, () => {
    console.log('=' .repeat(60));
    console.log(`  🟢 ZK REVENUE OPS NODE DEV SERVER ACTIVE ON PORT ${PORT}`);
    console.log(`  👉 Master Console:  http://localhost:${PORT}/index.html`);
    console.log(`  👉 REN Portal:      http://localhost:${PORT}/portal.html?ren=REN-001`);
    console.log('=' .repeat(60));
});

File: 05_Systems/Scripts/start-localhost-server.py
"""
---
Title: Localhost Dev Server (SYS-035)
ID: SYS-035
Type: Script (Python HTTP Server)
Module: 05_Systems/Scripts
BU: ZK Revenue Ops
Status: Active
Version: 1.0
Created: 2026-08-08
Updated: 2026-08-08
Owner: Zubair (zubairisa10@gmail.com)
---
"""

import http.server
import socketserver
import os
import sys

# Force UTF-8 output on Windows
if sys.stdout.encoding.lower() != 'utf-8':
    sys.stdout.reconfigure(encoding='utf-8')

PORT = 8080
DIRECTORY = r"C:\Users\Dell\Documents\Projects ZK Nexus"

class CustomHandler(http.server.SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory=DIRECTORY, **kwargs)

    def log_message(self, format, *args):
        # Quiet clean logging
        print(f"[LOCALHOST :{PORT}] {args[0]}")

if __name__ == "__main__":
    os.chdir(DIRECTORY)
    with socketserver.TCPServer(("", PORT), CustomHandler) as httpd:
        print("="*60)
        print(f"  ZK REVENUE OPS LOCALHOST DEV SERVER IS ACTIVE!")
        print(f"  Master Console:  http://localhost:{PORT}/index.html")
        print(f"  Client Portal:   http://localhost:{PORT}/portal.html")
        print("="*60)
        try:
            httpd.serve_forever()
        except KeyboardInterrupt:
            print("\nServer stopped.")
            sys.exit(0)
